### ISSUE TITLE
Allow parsing admonitions with capital letters, or document that only lowercase is allowed

### ISSUE THREAD

**Issue by kescobo** [JuliaLang/julia#33625]
In the `Markdown` stdlib, admonitions like `!!! note` produce nice call-outs (documented [here](https://docs.julialang.org/en/v1/stdlib/Markdown/#Admonitions-1)). At the moment, 

```
!!! note
    Nice note!

!!! weirdo
    Still nice

!!! Weirdo
    Booo!

!!! weirdO
    👎
```

Gets rendered as 

<img width="800" alt="image" src="https://user-images.githubusercontent.com/3502975/67236230-de724780-f416-11e9-976a-869c9ede2b32.png">

at least by `Documenter.jl`. This is especially confusing because the labeles for the block get rendered with captital letters.

The relevant [regex is here I think](https://github.com/JuliaLang/julia/blob/master/stdlib/Markdown/src/Common/block.jl#L210-L213). Could that be changed to `[a-zA-Z]+`, or else could the fact that only lowercase is allowed be documented above? Either way, I'm happy to make a PR.

**Comment by fredrikekre** [comment]
You can use another tag and have the "weird" title as the title
```
julia> md"""
       !!! tag "Weirdo"
           Booo!
       """
  │ Weirdo
  │
  │  Booo!
```

**Comment by kescobo** [comment]
Yes, certainly - That feature is great, well documented, and also allows spaces and punctuation. The issue is not so much that I can't get to the display I want, it's more that I just found it a bit confusing at first that `foo` gets rendered as `"Foo"` but that `Foo` in the markdown doesn't work at all, and couldn't find any explanation about it. 

I just went back and checked, and it looks like the fact that only lowercase letters is allowed was added by @mortenpi in #36589, so AFAIC, that closes this issue, thanks!
